client/src/components/EditForm.js

```diff
@@ -0,0 +1,191 @@
+import { useState } from 'react';
+
+const EditForm = (props) => {
+    // Initial student in case that you want to update a new student
+    const {
+      initialPost = {
+        id: null,
+        username: "",
+        imageurl: "",
+        alt: "",
+        dish: "",
+        restaurant: "",
+        content: "",
+        city: "",
+        date: "",
+      },
+    } = props;
+  
+    // We're using that initial student as our initial state
+    const [post, setPost] = useState(initialPost);
+  
+    //create functions that handle the event of the user typing into the form
+    const handleUsernameChange = (event) => {
+      const username = event.target.value;
+      setPost((post) => ({ ...post, username }));
+    };
+  
+    const handleImageURLChange = (event) => {
+      const imageurl = event.target.value;
+      setPost((post) => ({ ...post, imageurl }));
+    };
+  
+    const handleAltChange = (event) => {
+      const alt = event.target.value;
+      setPost((post) => ({ ...post, alt }));
+    };
+  
+    const handleDishChange = (event) => {
+      const dish = event.target.value;
+      setPost((post) => ({ ...post, dish }));
+    };
+  
+    const handleRestaurantChange = (event) => {
+      const restaurant = event.target.value;
+      setPost((post) => ({ ...post, restaurant }));
+    };
+  
+    const handleContentChange = (event) => {
+      const content = event.target.value;
+      setPost((post) => ({ ...post, content }));
+    };
+  
+    const handleCityChange = (event) => {
+      const city = event.target.value;
+      setPost((post) => ({ ...post, city }));
+    };
+  
+    const handleDateChange = (event) => {
+      const date = event.target.value;
+      setPost((post) => ({ ...post, date }));
+    };
+  
+    //A function to handle the post request
+    const makePost = (newPost) => {
+      return fetch("/api/blogposts", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify(newPost),
+      })
+        .then((response) => {
+          return response.json();
+        })
+        .then((data) => {
+          console.log("From the post ", data);
+          props.savePost(data);
+        });
+    };
+  
+    //a function to handle the Update request
+    const updatePost = (existingPost) => {
```

Comment:
This function is also on the `Dallas` page. Is this meant to be consolidated here?

Reply:
the editForm isn't complete, so for now it is a copy of Dallas until I figure out what is actually needed in here vs the original